File: frontend/services/authService.ts
import type {
  CancelDeletionRequest,
  ForgotPasswordRequest,
  LoginRequest,
  LogoutRequest,
  RegisterRequest,
  RefreshRequest,
  ResetPasswordRequest,
  TokenPairResponse,
  UserResponse,
  VerifyEmailResponse,
} from '../types/api.types';
import { apiClient } from './apiClient';

/** Endpoints de autenticação (03_CONTRATOS_API.md §2) — todos públicos. */
export const authService = {
  register(payload: RegisterRequest): Promise<UserResponse> {
    return apiClient
      .post<UserResponse>('/auth/register', payload)
      .then((response) => response.data);
  },

  login(payload: LoginRequest): Promise<TokenPairResponse> {
    return apiClient
      .post<TokenPairResponse>('/auth/login', payload)
      .then((response) => response.data);
  },

  refresh(payload: RefreshRequest): Promise<TokenPairResponse> {
    return apiClient
      .post<TokenPairResponse>('/auth/refresh', payload)
      .then((response) => response.data);
  },

  logout(payload: LogoutRequest): Promise<void> {
    return apiClient.post('/auth/logout', payload).then(() => undefined);
  },

  verifyEmail(token: string): Promise<VerifyEmailResponse> {
    return apiClient
      .get<VerifyEmailResponse>(`/auth/verify-email/${token}`)
      .then((response) => response.data);
  },

  forgotPassword(payload: ForgotPasswordRequest): Promise<void> {
    return apiClient.post('/auth/forgot-password', payload).then(() => undefined);
  },

  resetPassword(payload: ResetPasswordRequest): Promise<void> {
    return apiClient.post('/auth/reset-password', payload).then(() => undefined);
  },

  cancelDeletion(payload: CancelDeletionRequest): Promise<void> {
    return apiClient.post('/auth/cancel-deletion', payload).then(() => undefined);
  },
};

File: backend/src/main/java/com/irontrack/api/services/AuthService.java
package com.irontrack.api.services;

import com.irontrack.api.dto.request.ForgotPasswordRequest;
import com.irontrack.api.dto.request.LoginRequest;
import com.irontrack.api.dto.request.RefreshRequest;
import com.irontrack.api.dto.request.RegisterRequest;
import com.irontrack.api.dto.request.ResetPasswordRequest;
import com.irontrack.api.dto.response.AuthTokensResponse;
import com.irontrack.api.dto.response.MessageResponse;
import com.irontrack.api.dto.response.UserResponse;
import com.irontrack.api.dto.response.VerifyEmailResponse;

/**
 * 07_ROADMAP_BACKEND.md §C.1, item 3 — 03_CONTRATOS_API.md §2.1-§2.7.
 */
public interface AuthService {

    UserResponse register(RegisterRequest request);

    AuthTokensResponse login(LoginRequest request);

    AuthTokensResponse refresh(RefreshRequest request);

    void logout(RefreshRequest request);

    VerifyEmailResponse verifyEmail(String rawToken);

    MessageResponse forgotPassword(ForgotPasswordRequest request);

    void resetPassword(ResetPasswordRequest request);
}
